### DISCUSSION TITLE
Deploying an Appwrite/Database/Flutter app at my clients

### DISCUSSION THREAD

**Discussion by natgross** [appwrite/appwrite#3602] [General]
OK. So I get Appwrite, create a Flutter-Android project, [also] start a database and code away.
The day of reckoning comes, and its time to deploy my app to end-users. Now, for my customers that do NOT want their data on a cloud, they want an in-house db server, how do I proceed? These customers don't trust Google's Firebase either. So, until now, using MongoDB as the server, no problem. We install MongoDB on their inhouse server, point my software to their server, done. (These folks will not touch Mongo Atlas either.)
So, the question is, can an Appwrite database be deployed at end-user's place, like with other databases? (And my software interact with it?)

Thanks;
-nat

**Comment by stnguyen90** [comment]
Yes, it's possible to use an external database. It should be a matter of updating the [environment variables](https://appwrite.io/docs/environment-variables#mariadb). 

They may want to spin up a self hosted server to maintain the storage files too.

**Reply by natgross** [reply]
Sounds great. Let me see if I understand this in baby steps.
1. As the developer, I write the code and access my Appwrite server and services with a Unique Project Id.
2. At my customer site, I have them install the Appwrite server? 
2b. They register as a developer?
2c. They need to create a project with the same name as mine?

**Reply by stnguyen90** [reply]
Pretty much yes. You can also use the [Appwrite CLI](https://appwrite.io/docs/command-line#gettingStarted) to help with copying the collections from your dev instance of Appwrite to their production instance.
